.github/CONTRIBUTING.md

```diff
@@ -81,6 +98,29 @@ $ make html
 
 The built documentation can then be found in `docs/_build/html/`.
 
+To file a pull request, create a new branch on top of the upstream repository:
+
+```console
+$ git fetch --all
+$ git checkout -b my_topical_branch upstream/main
+```
+
+Make your changes, push them to your fork (the remote *origin*):
+
+```console
+$ git push -u origin
```

Comment:
FYI What a push without a branch name does depends on the Git configuration and the defaults depend on the Git version. Before Git 2.0, it'd push all the local branches by default, not just the current one: https://stackoverflow.com/a/948397.
Though, it's probably safe to keep this instruction as is.

`-u` creates a "tracking branch" connection between the local branch and said remote, saving this information in the repo-local Git config.